src/intersections/lines.jl

```diff
@@ -43,14 +28,30 @@ end
 function intersectpoint(l1::Line, l2::Line)
   a, b = l1(0), l1(1)
   c, d = l2(0), l2(1)
-
-  v1  = a - b
-  v2  = c - d
-  v12 = a - c
-
-  # the intersection point lies in between a and b at a fraction s
-  # (https://en.wikipedia.org/wiki/Line-line_intersection#Formulas)
-  s = (v12[1] * v2[2] - v12[2] * v2[1]) / (v1[1] * v2[2] - v1[2] * v2[1])
-
-  a - s*v1
+  p₁, _ = intersectparameters(a, b, c, d)
+  a + p₁ * (b - a)
+ end
+
+# compute the parameters of the lines defined by the points a -> b and c -> d 
+# so that the resulting points have minimal distance
+# the ranks help to identify the different types of intersection
+function intersectparameters(a::Point{N,T}, b::Point{N,T}, c::Point{N,T}, d::Point{N,T}) where {N,T}
+  # solves the equation (approximately): a + λ₁ ⋅ v⃗₁ = c + λ₂ ⋅ v⃗₂, v⃗₁ = b - a, v⃗₂ = d - c
+  A = [(b - a) (c - d)]
+  b = c - a
+
+  λ = A \ b
+
+  # calculate the rank of the augmented matrix
+  rₐ = rank([A b], atol = atol(T))
+  # calculate the rank of the rectangular matrix
+  r = rank(A, atol = atol(T))
+
+  # Intersection r == rₐ == 2
+  # Collinear: r == rₐ == 1
+  # no intersection: r != rₐ
+  #   no intersection and parallel:  r == 1, rₐ == 2
+  #   no intersection, skew lines: r == 2, rₐ == 3
+
+  λ[1], λ[2], r, rₐ
```

Comment:
i don't know a situation where both values are needed in a vector, so when applying this function one would always have to access the individual elements. It will make code using this helper function more readable if we keep lambda[1] and lambda[2] seperated